CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+Contributing to SDM Schemas
+===========================
+
+Creating a PR
+-------------
+
+All changes to the repository must be done through the standard GitHub PR process. Direct changes to the `main` branch are disallowed by branch protection rules.
+
+The PR title should be formatted like: `[TICKET]: [DESCRIPTION]` where `TICKET` is a Jira ticket, usually from Data Management, so `DM-#####`.
+
+For example, this is a valid PR title (with a dummy ticket number):
+
+`DM-12345: Add ra and dec columns`.
+
+The description of the PR should contain clear information on what is being changed and why.
+
+An appropriate reviewer should be assigned by the PR author based on which schema is being changed. If this is not known, [JeremyMcCormick](https://github.com/JeremyMcCormick) can be used as a default.
+
+Schema Versioning
+-----------------
+
+A pull request may need to include an update to the schema's version.
+
+Usage of semantic versioning is recommended, which defines a version like `MAJOR.MINOR.PATCH`, such as `1.2.3`.
+
+The following guidelines should be used for this versioning scheme:
+
+- Different `MAJOR` versions are not compatible.
+- Different `MINOR` versions are backward compatible.
+- Different `PATCH` versions are completely compatible.
+
+Backward compatibility implies that a client using an interface derived from a schema with a higher `MINOR` version number will be able to read and write into a database with a lower `MINOR` version.
+But a client using a lower `MINOR` version number will not be compatible with a database that contains the changes from a higher version.
+
+There may be cases where significant schema changes may still be compatible if the client can handle them transparently.
+In general, version changes will need to be evaluated on a case-by-case basis and may depend on the particular schema being updated.
+
+Typical Schema Changes
+----------------------
+
+Here are some guidelines for incrementing the version number based on what changes are being performed:
+
+| Change | Increment | Notes |
+|--------|-----------|-------|
+| Adding a table | `MAJOR` | Clients will want to access the new table, so this is backward incompatible.
+| Adding a column | `MAJOR` | Clients will want to access the new column, so this is incompatible with the existing schema.|
+| Removing a column | `MINOR` | Clients should not care about a missing column, so this should be backward compatible.|
```

Comment:
These seem a bit broader and harder to visualize than the original text. Perhaps re-emphasize that this is talking about _new_ clients that already have had references to the column added/removed?

Actually, I think that may be the key issue. The original docs were specific to the APDB, which is accessed almost exclusively through the `dax_apdb` API. Does the same go for other tables, or is there more room for e.g. references to a removed column?

Reply:
> These seem a bit broader and harder to visualize than the original text. Perhaps re-emphasize that this is talking about _new_ clients that already have had references to the column added/removed?
> 
> Actually, I think that may be the key issue. The original docs were specific to the APDB, which is accessed almost exclusively through the `dax_apdb` API. Does the same go for other tables, or is there more room for e.g. references to a removed column?

I can clarify this text. You're completely right that a client would need to be updated if a column was dropped from the schema and it was being accessed explicitly.

Reply:
The other option is that the guidelines/rules for the versioning could go under the APDB-specific documentation, though I would like to provide some general recommendations for any versioned schema, if possible.

Reply:
I moved these guidelines to `docs/APDB.md` as they are APDB-specific.

Reply:
> Some of these I'm not even sure I agree with - wouldn't removing a column be a backward-incompatible change necessitating a MAJOR version bump since old clients could be attempting to access it?

In this context @andy-slac is specifically defining "backward compatible" as "new client with old DB". I agree it's confusing, but it's at least internally consistent.

Reply:
> > Some of these I'm not even sure I agree with - wouldn't removing a column be a backward-incompatible change necessitating a MAJOR version bump since old clients could be attempting to access it?
> 
> In this context @andy-slac is specifically defining "backward compatible" as "new client with old DB" (that definition seems to have stayed in the generic docs and not the APDB-specific ones?). I agree it's confusing, but it's at least internally consistent.

Okay, thank you for clarifying. This is usually defined the other way around, but I moved this into the APDB-specific docs so that it was clear this is not a general rule for versioning all schemas.

Reply:
> that definition seems to have stayed in the generic docs and not the APDB-specific ones?

Good catch - I'm going to move this under the APDB docs as well. Each schema which uses versioning is probably going to handle things differently.